### ISSUE TITLE
Windows mobile support

### ISSUE THREAD

**Issue by tomchiverton** [adnathan/ace#5]

**Comment by adnathan** [comment]
That's a great one, thanks!  Supporting this is made more interesting by the fact that the Cordova host app for Windows is a WWA rather than a XAML app hosting a WebView, but this is definitely something I want to support.

**Comment by pkunze** [comment]
IMO this could also make ace a Mayor Player in Cross Platform dev. I dont know any other tools that provide such a feature (xamarin set aside)

**Comment by felquis** [comment]
This will be great!

I worked with Ionic a few months ago, and I gave up of Cordova/Ionic because of the ecosystem, the difficult access to native features and cordova plugins that doesn't work well, but looking these example, this tool is such a piece of candy. Hope it get better and better, as a utility tool.

**Comment by PaulOst** [comment]
Did you guys forget about Windows 10?

**Comment by adnathan** [comment]
Not at all. :)  I would love to have a Windows implementation, but I haven't gotten to it yet.

**Comment by PaulOst** [comment]
Thanks for the reply, Adam!

**Comment by PaulOst** [comment]
Adam, when adding support for Windows I think you only need to target UWP apps. Doing so will let us target this new product with our Enterprise apps: [http://mspoweruser.com/panasonics-toughpad-fz-f1-windows-10-tablet-announced-at-mwc/](url)

**Comment by tomchiverton** [comment]
## I think most consumers will have WP8 for some time. WP10 is not penetrating fast at all, though some of our customers do ask about the possibility of support for it in future 

Tom

**Comment by PaulOst** [comment]
Most consumers will have WP8 for some time? WP8 is at 1.5% share and falling while Windows 10 is on the rise. Adam, please release support for UWP first before supporting older variations of Windows.

**Comment by tomchiverton** [comment]
We may be at cross purposes, but as Windows only has 2.8% of the market 
http://www.winbeta.org/news/windows-phones-fall-2-8-u-s-market-share-according-comscore
then 1.5% on WP8 is a large fraction. WP10 is not exactly selling like hot cakes, and few people see the need to upgrade from WP8 to it. 

## If we are talking desktop, then yes, I think only Win7 and later are relevant. A lot of enterprises are no where near upgrading their Win7 fleets. 

Tom
Sent from my phone.

**Comment by PaulOst** [comment]
Agreed, Tom. My feedback is that project Ace first supports W10 UWP. I think that's the future of Windows not the dying W8.x, WP8 platforms. The apps my company would build using project ACE wouldn't be the kind that would target [or be beneficial on] the Windows 7 desktop.

**Comment by aloisdg** [comment]
Universal Apps would be  great!

**Comment by adnathan** [comment]
I do plan on looking at UWP first. But I should reiterate that I'm happy for others to contribute as well! :)

**Comment by aloisdg** [comment]
@adnathan I have my own commitments (ok I am proud of this one). One day, why not? :)

**Comment by Li-Yanzhi** [comment]
@adnathan just know this project from MSDN Channel9, Great idea & project,  with Cordova + ACE, it seems that everything is possible!

**Comment by hgultekin** [comment]
I was just writing a similar app and looking for a native tab-bars for android, ios and windows phone 10. Unfortunately none of them has support for wp10. Even Microsoft ace :(. Please add support for wp10. 

Another good feature would be rendering it html5 as well. Use case: One code runs on android, wp10, ios and web. 

I see that compiler part is closed code. How can I contribute? I would love to be part of it #

**Comment by tomchiverton** [comment]
## Ionic 2 (via Cordova / PhoneGap for mobile) does that. Currently beta though. 

Tom

**Comment by ztl8702** [comment]
@tomchiverton Yeah, but that's HTML not native UI.

**Comment by professoroakz** [comment]
RIP WP :/

**Comment by ztl8702** [comment]
@OktayGardener WP is dying, but not Win10 Mobile and UWP.

**Comment by hemrika** [comment]
ace.platform == "Windows"  

Would be awesome !! Even just invoke custom code...

**Comment by adamiprinciples** [comment]
+1 for this. Even just basic support would be great. I don't mind writing controls myself, just some POC's for windows would be good enough.

**Comment by kkirby** [comment]
Yeah, this would make my day ... even my year.
